### ISSUE TITLE
Not possible to make beeping interrupt `:normal` when run from `:silent!`

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#10928]
When run from `:silent!`, beeping does not flush typeahead, and therefore does not interrupt `:normal`. This makes `:normal` behave differently, and it is hard to make `:normal` behave the same as when run without `:silent!`.

An easy potential solution is to flush typeahead on beeping even if `:silent!` is used, but there may be people who rely on the current behavior, and if this is changed then there is no longer a way to achieve the current behavior.

Other solutions may require adding new commands, I guess.

**Comment by brammool** [comment]
> When run from `:silent!`, beeping does not flush typeahead, and
> therefore does not interrupt `:normal`. This makes `:normal` behave
> differently, and it is hard to make `:normal` behave the same as when
> run without `:silent!`.
> 
> An easy potential solution is to flush typeahead on beeping even if
> `:silent!` is used, but there may be people who rely on the current
> behavior, and if this is changed then there is no longer a way to
> achieve the current behavior.
> 
> Other solutions may require adding new commands, I guess.

Yes, in some corner cases using ":silent!" causes the error to be
ignored and what follows to be executed differently.  It is hard to say
if this is intentional or not.  It can lead to weird problems.
Especially in Vim9 code, which often depends on aborting execution when
an error is encountered (I recently fixed a bug where a variable was
created but not set, because the RHS had an error, but then continuing
would use the invalid variable value).

We could make a difference between "be quiet about errors but behave
like it did happen" and "completely ignore the error".  Although the
latter can't always be done, the error might have side effects.  Anyway,
problem is that any change would have backwards compatibility problems.
And it's going to be tricky to implement.

-- 
Q: How does a UNIX Guru pick up a girl?
A: look; grep; which; eval; nice; uname; talk; date;

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
